Infisical/infisical
# feat: email branding and accessibility

# Description 📣

<p style="text-align: center; font-weight: bold;">Backend</p>

<kbd>feat</kbd>

- Improved the email templates' Infisical branding
  - `emailMfa`
  - `emailVerification`
  - `newDevice`
  - `organizationInvitation`
  - `workspaceInvitation`
- Added extra substitutions to the email controllers
  - `title`
  - `description`
  - `year` (automatic)
- Changed the email title to quickly see what it's about
  - e.g. email verification code in the email subject (notification friendly)
- Redirecting the user directly to the invitation's project
  - `workspaceInvitation`
  - e.g. ``${getSiteURL()}/dashboard/${workspaceId}``
- Improved client compatibility/accessibility by providing both HTML + plain text email format
  - https://nodemailer.com/message/ `text:`
  - using this project : https://github.com/html-to-text/node-html-to-text
    - e.g. [before](https://github.com/html-to-text/node-html-to-text/blob/master/packages/html-to-text/test/test.html)/[after](https://github.com/html-to-text/node-html-to-text/blob/master/packages/html-to-text/test/test.txt)

<p style="text-align: center; font-weight: bold;">Frontend</p>

<kbd>fix</kbd>

- Typos in `password-reset` page

<p style="text-align: center; font-weight: bold;">Development</p>

<kbd>fix</kbd>

- YAML formatted environment variables definition docker-compose files, to align with the rest of the parameters

## Type ✨


## Screenshots

<table>
  <tr>
    <th>emailMfa</th>
    <th>emailVerification</th>
    <th>passwordReset</th>
    <th>newDevice</th>
    <th>organizationInvitation</th>
    <th>workspaceInvitation</th>
  </tr>
  <tr>
    <td>
      <img src="https://user-images.githubusercontent.com/72856427/226164759-97faf0eb-d566-42d8-8328-658475b072ab.png" />
    </td>
    <td>
      <img src="https://user-images.githubusercontent.com/72856427/226164617-786387c2-16c9-4c64-b8df-7adbdaa443f2.png" />
    </td>
    <td>
      <img src="https://user-images.githubusercontent.com/72856427/226164642-9d3f5367-c227-4e5b-afd4-d8bea033a3c7.png" />
    </td>
    <td>
      <img src="https://user-images.githubusercontent.com/72856427/226164743-089a8658-6431-43b6-9db6-5a6153a8fb99.png" />
    </td>
    <td>
      <img src="https://user-images.githubusercontent.com/72856427/226164662-c48968bf-9b53-4823-a633-035cc4968eab.png" />
    </td>
    <td>
      <img src="https://user-images.githubusercontent.com/72856427/226164669-74ad72c9-098d-4ade-8e5d-c36c886e6055.png" />
    </td>
  </tr>
</table>


Below a plain text example auto-generated with `html-to-text` from `newDevice`'s HTML template

```
Infisical [https://app.infisical.com/_next/image?url=%2Fimages%2Fbiglogo.png&w=128&q=75] https://infisical.com


NEW CONNECTION


WE'RE VERIFYING A RECENT LOGIN FOR TEST@INFISICAL.COM

TIMESTAMP    Sun Mar 01 2023 01:01:01 GMT+0000 (Coordinated Universal
             Time)
IP-ADDRESS   172.25.0.1
USER-AGENT   Mozilla/5.0 <redacted>

If it wasn't you or if you believe that this login is suspicious,
please contact us at team@infisical.com [team@infisical.com] or/and reset your password immediately!

Github [https://cdn.simpleicons.org/github/fcfdf7]

Twitter [https://cdn.simpleicons.org/twitter/fcfdf7]

Slack [https://cdn.simpleicons.org/slack/fcfdf7]

LinkedIn [https://cdn.simpleicons.org/linkedin/fcfdf7]

Copyright © 2023 Infisical Inc.

website [https://infisical.com/] docs [https://infisical.com/docs]
```

# Tests 🛠️

Please follow the https://infisical.com/docs/contributing/developing instructions to setup your local environment to test the changes, or do like you're used to do if you know what you're doing :v:

```sh
# .env

# Edit the below .env variables, if you want to use the local mailhog server
SMTP_HOST="smtp-server"
SMTP_FROM_ADDRESS="dev@infisical.local"
SMTP_USERNAME="dev@infisical.local"
SMTP_PASSWORD=""
```

```sh
# Start the development stack locally (build it with the new changes)
# using make : make up-dev
docker-compose up -f ./docker-compose.dev.yaml -d --build
```

Once the application is up and ready, here's the scenarios that will trigger the different emails :
1. Create a new account
     - `emailVerification` template
2. Log out/in to your account
     - `newDevice`
     - Enable MFA in your user settings
       - Log in and out to trigger `emailMfa`
3. Once logged in, invite another member to your organization (https://app.infisical.com/settings/org/<org-id>)
     - `organizationInvitation`
     - Then invite him to a given project (https://app.infisical.com/users/<project-id>)
       - `workspaceInvitation`

For each email, you can check the plain text version (e.g. on Mailhog you have a "Plain text" tab next to the "HTML" one)

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/code-of-conduct). 📝
